src/storage/v3/shard.cpp

```diff
@@ -1075,9 +1077,14 @@ bool Shard::CreateIndex(LabelId label, const std::optional<uint64_t> desired_com
 }
 
 bool Shard::CreateIndex(LabelId label, PropertyId property, const std::optional<uint64_t> desired_commit_timestamp) {
-  // TODO Fix Index
-  // if (!indices_.label_property_index.CreateIndex(label, property, labelspace.access())) return false;
-  return false;
+  if (label == primary_label_ && schemas_.GetSchema(primary_label_)->second.size() == 1 &&
+      schemas_.GetSchema(primary_label_)->second[0].property_id == property) {
+    // Index already exists on primary key
+    return false;
+  }
```

Comment:
Based on this logic it is possible to create an index on a secondary label + primary property. Because the indices are only "checked against" in label and property updates, it is possible to construct a vertex with predefined secondary labels. In that case the vertex will be left out from the index if I am not mistaken. Do we have any test to check this exact use case? If it complicates things to much, then we can "fall back" to the old way of creating a vertex: only primary label (not even that after the other PR of yours) and the primary key is provided, everything else has to be set using `SetProperties` or `AddLabel`.

Reply:
Will cover this with test cases